e2e/aws/rds_test.go

```diff
@@ -0,0 +1,224 @@
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	apidefaults "github.com/gravitational/teleport/api/defaults"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/integration/helpers"
+	"github.com/gravitational/teleport/lib/auth"
+	"github.com/gravitational/teleport/lib/client"
+	"github.com/gravitational/teleport/lib/defaults"
+	"github.com/gravitational/teleport/lib/service"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/teleport/lib/srv/alpnproxy"
+	alpncommon "github.com/gravitational/teleport/lib/srv/alpnproxy/common"
+	"github.com/gravitational/teleport/lib/tlsca"
+	"github.com/gravitational/teleport/lib/utils"
+)
+
+func TestDatabases(t *testing.T) {
+	// get test settings
+	awsRegion := mustGetEnv(t, awsRegionEnv)
+	dbDiscoverySvcRoleARN := mustGetEnv(t, dbDiscoverySvcRoleARNEnv)
+	dbSvcRoleARN := mustGetEnv(t, dbSvcRoleARNEnv)
+	dbUser := mustGetEnv(t, dbUserEnv)
+	rdsPostgresInstanceName := mustGetEnv(t, rdsPostgresInstanceNameEnv)
+
+	cluster := createTeleportCluster(t,
+		withSingleProxyPort(t),
+		withDiscoveryService(t, "db-e2e-test", types.AWSMatcher{
+			Types:   []string{services.AWSMatcherRDS},
+			Tags:    mustGetDiscoveryMatcherLabels(t),
+			Regions: []string{awsRegion},
+			AssumeRole: &types.AssumeRole{
+				RoleARN: dbDiscoverySvcRoleARN,
+			},
+		}),
+		withDatabaseService(t, services.ResourceMatcher{
+			Labels: types.Labels{types.Wildcard: {types.Wildcard}},
+			AWS: services.ResourceMatcherAWS{
+				AssumeRoleARN: dbSvcRoleARN,
+			},
+		}),
+		withFullDatabaseAccessUserRole(t),
+	)
+
+	wantDBNames := []string{
+		rdsPostgresInstanceName,
+	}
+	// wait for the databases to be discovered
+	waitForDatabases(t, cluster.Process, wantDBNames...)
+	// wait for the database heartbeats from database service
+	waitForDatabaseServers(t, cluster.Process, wantDBNames...)
+
+	rdsPostgresInstance := tlsca.RouteToDatabase{
+		ServiceName: rdsPostgresInstanceName,
+		Protocol:    defaults.ProtocolPostgres,
+		Username:    dbUser,
+		Database:    "postgres",
+	}
+	tests := []struct {
+		name             string
+		route            tlsca.RouteToDatabase
+		testDBConnection dbConnectionTestFunc
+	}{
+		{
+			name:             "RDS postgres instance",
+			route:            rdsPostgresInstance,
+			testDBConnection: postgresConnTestFn(cluster),
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+			defer cancel()
+			test.testDBConnection(t, ctx, test.route)
+		})
+	}
+}
+
+type dbConnectionTestFunc func(*testing.T, context.Context, tlsca.RouteToDatabase)
+
+// postgresConnTestFn tests connection to a postgres database.
+func postgresConnTestFn(cluster *helpers.TeleInstance) dbConnectionTestFunc {
+	return func(t *testing.T, ctx context.Context, route tlsca.RouteToDatabase) {
+		lp := startLocalALPNProxy(t, ctx, cluster, route)
```

Comment:
Can't you reuse the logic from https://github.com/gravitational/teleport/blob/1178971b3d7c6f82d93c476b45c1c9ebd3b5487f/integration/db/db_integration_test.go#L194-L207

Reply:
I can add a test for direct connection to the cluster proxy reusing that func, but still would keep what I have to test the local proxy path

Reply:
added direct conn test as well [ddce196](https://github.com/gravitational/teleport/pull/29755/commits/ddce19696d334e59a3e51e7d09199f06543868b2)